src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Routing/CatchAll.razor

```diff
@@ -0,0 +1,25 @@
+﻿@page "/routing/catch-all/{*parameter}"
```

Comment:
Why are these new test cases being added to `RazorComponents`? If I understand correctly, the new routing behaviors are being applied to all Blazor apps, so I would think the E2E tests should be in `BasicTestApp` so the tests will run against both Server and WebAssembly.

Reply:
We want to test using the routing system in ASP.NET Core when prerendering, and the server routing system when we then render interactively.

If we put them in BasicTestApp they would only test the interactive routing piece.

Reply:
I've filed https://github.com/dotnet/aspnetcore/issues/49845 to add a flavor for this in RC2 on webassembly